### PR TITLE
Vellum case management migration ui

### PR DESCRIPTION
**Pull request by orangejenny**
UI that @dimagi/product can use to manage case management migration. Planning to add ability to actually migrate domains from this page. Would be nice to get this deployed before first wave of migrations Monday.

screen shot 2016-07-15 at 2 17 30 pm

@benrudolph

### CODE REVIEW

File: corehq/apps/hqadmin/views.py [hunk 1]
```diff
@@ -4,7 +4,8 @@
 import json
 import socket
 import csv
-from datetime import timedelta, date
+import os
```
Comment (dimagimon) [inline corehq/apps/hqadmin/views.py]: F401 'os' imported but unused

File: corehq/apps/hqadmin/views.py [hunk 2]
```diff
@@ -458,6 +460,43 @@ def page_context(self):
...
 
 
+class VCMMigrationView(BaseAdminSectionView):
+    urlname = 'vcm_migration'
+    page_title = ugettext_lazy("Vellum Case Management Migration")
+    template_name = 'hqadmin/vcm_migration.html'
+
+    @property
+    def page_context(self):
+        domain_audit = namedtuple('domain_audit', 'name emailed migrated notes')
```
Comment (dimagimon) [inline corehq/apps/hqadmin/views.py]: F841 local variable 'domain_audit' is assigned to but never used

File: corehq/apps/hqadmin/views.py [hunk 3]
```diff
@@ -458,6 +460,43 @@ def page_context(self):
...
+            m.notes = self.request.POST['notes']
+            m.save()
+        else:
+            domains = self.request.POST['domains'].split(",")
+            for d in domains:
+                m = VCMMigration.objects.get(domain=d)
+                if action == 'email':
+                    send_mail_async.delay(
+                        _(u'Upcoming migration to easy references.'),
+                            'Upcoming migration to easy references. Two weeks from now.',
```
Comment (dimagimon) [inline corehq/apps/hqadmin/views.py]: E131 continuation line unaligned for hanging indent

File: corehq/apps/hqadmin/views.py [hunk 4]
```diff
@@ -458,6 +460,43 @@ def page_context(self):
...
+            m = VCMMigration.objects.get(domain=self.request.POST['domain'])
+            m.notes = self.request.POST['notes']
+            m.save()
+        else:
+            domains = self.request.POST['domains'].split(",")
+            for d in domains:
+                m = VCMMigration.objects.get(domain=d)
+                if action == 'email':
+                    send_mail_async.delay(
+                        _(u'Upcoming migration to easy references.'),
```
Comment (orangejenny) [inline corehq/apps/hqadmin/views.py]: @krsdimagi Can you send me the actual subject line & email copy? And also confirm that this email should come from `commcarehq-support@dimagi.com`?

**Comment by orangejenny** [comment]
@krsdimagi Can you send me the actual subject line & email copy? And also confirm that this email should come from `commcarehq-support@dimagi.com`?

File: corehq/apps/hqadmin/migrations/0002_vcmmigrationaudit.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
```
Comment (benrudolph) [inline corehq/apps/hqadmin/migrations/0002_vcmmigrationaudit.py]: nit: in the future you can combine these all into 1 before PRing by rolling back the migrations, deleting them, and then rerunning makemigrations

File: corehq/apps/hqadmin/models.py [hunk 6]
```diff
@@ -60,3 +62,14 @@ def get_by_pillow_name(cls, pillow_name):
...
+
+
+class VCMMigration(models.Model):
+    domain = models.CharField(max_length=255, null=False, unique=True)
+    emailed = models.DateTimeField(null=True)
+    migrated = models.DateTimeField(null=True)
+    notes = models.TextField(null=True)
+
+    @property
+    def admins(self):
```
Comment (benrudolph) [inline corehq/apps/hqadmin/models.py]: nit: could probably be memoized

**Comment by benrudolph** [comment]
👍 nthing blocking

File: corehq/apps/hqadmin/models.py (hunk 6 above)
Comment (orangejenny) [inline corehq/apps/hqadmin/models.py]: fixed

File: corehq/apps/hqadmin/migrations/0002_vcmmigrationaudit.py (hunk 5 above)
Comment (orangejenny) [inline corehq/apps/hqadmin/migrations/0002_vcmmigrationaudit.py]: cool, good to know

**Comment by benrudolph** [comment]
(feel free to self merge once product kevin signs off)

File: corehq/apps/hqadmin/views.py [hunk 7]
```diff
@@ -458,6 +459,49 @@ def page_context(self):
         return context
 
 
+class VCMMigrationView(BaseAdminSectionView):
+    urlname = 'vcm_migration'
+    page_title = ugettext_lazy("Vellum Case Management Migration")
```
Comment (millerdev) [inline corehq/apps/hqadmin/views.py]: I understand why, but I still feel a little weird about using the name "Vellum" in HQ copy. It's a code name used privately by the dev team. I guess putting it on an admin page is fine...

**Comment by millerdev** [comment]
👍 from me

**Comment by krsdimagi** [comment]
Hey @orangejenny here is a link to the copy: https://docs.google.com/document/d/13HKq28xppLOdEvRxn8zLULo0bZtWqN4FV2qhY8CWjbQ/edit

commcare-support@dimagi.com is fine.

File: corehq/apps/hqadmin/views.py (hunk 7 above)
Comment (orangejenny) [inline corehq/apps/hqadmin/views.py]: Yeah, I see your point, hadn't thought about that. I think I still favor referencing "vellum" over calling it "Case Management in the Form Builder Migration" for the sake of space.

**Comment by orangejenny** [comment]
Updated copy, should be good to go.

**Pull request merged**